### PR TITLE
Add a documentation page about the block editor settings

### PR DESCRIPTION
**Pull request by youknowriad**
Related #53874

## What?

Just documents some of the most important settings a user can pass as a prop to the `BlockEditorProvider` component. I think we're still missing the "theme.json" settings and styles from this list but these should be added when we decided on a stable name.

### Test the documentation website

```
cd platform-docs
npm install
npm start
```

### CODE REVIEW

File: platform-docs/docs/basic-concepts/settings.md [hunk 1]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+			ol {
+				margin: 0;
+				padding: 0;
+			}
+
+			ul li,
+			ol li {
+				margin-bottom: initial;
+			}
+
```
Comment (ntsekouras) [inline platform-docs/docs/basic-concepts/settings.md:28]: Nit: we can add fewer css rules in the example.

**Review by ntsekouras — APPROVED** [review]
LGTM, thanks!

File: platform-docs/docs/basic-concepts/settings.md [hunk 2]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+				max-width: none;
+			}
+		`,
+	},
+];
+
+export default function App() {
+	return (
+		<BlockEditorProvider settings={ { styles: editorStyles } }>
+			<Editor />
```
Comment (t-hamano) [inline platform-docs/docs/basic-concepts/settings.md]: I was confused whether the `Editor` component was expected to be imported from a WordPress package or whether it was its own component. How about using `<BlockCanvas />` component to match other documents?

**Review by t-hamano — APPROVED** [review]

File: platform-docs/docs/basic-concepts/settings.md [hunk 3]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+		</BlockEditorProvider>
+	);
+}
+```
+
+## mediaUpload
+
+Some core blocks like the image or video blocks allow users to render media within the block editor, by default you can use external URLs but if you want to allow users to upload media from their computer you need to provide a `mediaUpload` function. Here's a quick example of such function:
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
Some core blocks, like the image or video blocks, allow users to render media within the block editor. By default, you can use external URLs but if you want to allow users to upload media from their computer, you need to provide a `mediaUpload` function. Here's a quick example of such function:
```

Nit: Minor tweaks to punctuation to improve reading flow.

File: platform-docs/docs/basic-concepts/settings.md [hunk 4]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+    },
+    mediaType: 'image',
+    async fetch( query = {} ) {
+        const defaultArgs = {
+            mature: false,
+            excluded_source: 'flickr,inaturalist,wikimedia',
+            license: 'pdm,cc0',
+        };
+        const finalQuery = { ...query, ...defaultArgs };
+        // Sometimes you might need to map the supported request params according to `InserterMediaRequest`.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
        // Sometimes you might need to map the supported request params according to the `InserterMediaRequest`
```

File: platform-docs/docs/basic-concepts/settings.md [hunk 5]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+    },
+    getReportUrl: ( { sourceId } ) =>
+        `https://wordpress.org/openverse/image/${ sourceId }/report/`,
+    isExternalResource: true,
+}
+```
+
+## hasFixedToolbar
+
+Whether or not the `BlockTools` component renders the toolbar in a fixed position or if follows the position of the selected block. Defaults to `false`.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
Whether or not the `BlockTools` component renders the toolbar in a fixed position or follows the position of the selected block. Defaults to `false`.
```

File: platform-docs/docs/basic-concepts/settings.md [hunk 6]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+
+## focusMode
+
+The focus mode is a special mode where any block that is not selected is shown with a reduced opacity, giving more prominence to the selected block. Defaults to `false`.
+
+## keepCaretInsideBlock
+
+By default arrow keys can be used to navigate accross blocks. You can turn off that behavior by and keep the caret inside the currently selected block using this setting. Defaults to `false`.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
By default, arrow keys can be used to navigate across blocks. You can turn off that behavior and keep the caret inside the currently selected block using this setting. Defaults to `false`.
```

**Review by aaronrobertshaw — COMMENTED** [review]
These docs are looking great, nice work 👍 

I spotted a couple of typos so have left inline comments for those and some other minor tweaks

File: platform-docs/docs/basic-concepts/settings.md [hunk 7]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+		}
+	}
+}
+```
+
+Providing a `mediaUpload` function also enables drag and dropping files into the editor to upload them.
+
+## inserterMediaCategories
+
+The inserter media categories setting is an array of media categories to display in the inserter. Each category is an object with a `name`, `labels` a `fetch` function and a few extra keys. Example:
```
Comment (ramonjd) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
The inserter media categories setting is an array of media categories to display in the inserter. Each category is an object with `name` and `labels` values, a `fetch` function and a few extra keys. Example:
```

File: platform-docs/docs/basic-concepts/settings.md (hunk 5 above)
Comment (ramonjd) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
Whether the `BlockTools` component renders the toolbar in a fixed position or if follows the position of the selected block. Defaults to `false`.
```

File: platform-docs/docs/basic-concepts/settings.md [hunk 8]
```diff
@@ -3,3 +3,187 @@ sidebar_position: 4
...
+
+By default arrow keys can be used to navigate accross blocks. You can turn off that behavior by and keep the caret inside the currently selected block using this setting. Defaults to `false`.
+
+## codeEditingEnabled
+
+Allows the user to edit the currently selected block using an HTML code editor. Note that using the code editor is not supported by all blocks and that it has the potential to create invalid blocks depending on the markup the user provides. Defaults to `true`.
+
+## canLockBlocks
+
+Whether or not the user can lock blocks. Defaults to `true`.
```
Comment (ramonjd) [inline platform-docs/docs/basic-concepts/settings.md]:
```suggestion
Whether the user can lock blocks. Defaults to `true`.
```

**Review by ramonjd — COMMENTED** [review]
LGTM!

Left some predictably pedantic suggestions. 

"Whether or not" is interesting though - it's mostly used when the outcome is not important. Like, _"Whether or not Github allows threads in comments, I'm still going to use it!"_  :trollface:

**Pull request merged**